src/Akeneo/Channel/back/Infrastructure/Query/Sql/SqlGetChannels.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+
+namespace Akeneo\Channel\Infrastructure\Query\Sql;
+
+use Akeneo\Channel\API\Query\Channel;
+use Akeneo\Channel\API\Query\GetChannels;
+use Akeneo\Channel\API\Query\LabelCollection;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license   https://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class SqlGetChannels implements GetChannels
+{
+    private Connection $connection;
+    private ?array $cache = null;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    public function findAll(): array
+    {
+        if (null === $this->cache) {
+            $sql = <<<SQL
+                SELECT 
+                    c.code AS channelCode, 
+                    JSON_OBJECTAGG(l.id, l.code) AS localeCodes,
+                    JSON_OBJECTAGG(ct.locale, ct.label) AS labels,
+                    JSON_OBJECTAGG(cur.id, cur.code) AS activatedCurrencies
+                FROM pim_catalog_channel c
+                LEFT JOIN pim_catalog_channel_locale cl 
+                    ON c.id = cl.channel_id
+                LEFT JOIN pim_catalog_locale l 
+                    ON cl.locale_id = l.id
+                LEFT JOIN pim_catalog_channel_translation ct 
+                    ON c.id = ct.foreign_key
+                LEFT JOIN pim_catalog_channel_currency cc 
+                    ON c.id = cc.channel_id
+                LEFT JOIN pim_catalog_currency cur 
+                    ON cc.currency_id = cur.id AND cur.is_activated = 1
```

Comment:
You didn't need to check is_activated as we check via pim_catalog_channel_currency
```suggestion
                    ON cc.currency_id = cur.id
```